packages/common-soroban/src/project/models.ts

```diff
@@ -14,12 +14,99 @@ import {
   SubqlCustomDatasource,
   FileReference,
   CustomDataSourceAsset,
-  SubqlEventHandler,
+  SorobanBlockFilter,
+  SorobanTransactionFilter,
+  SorobanOperationFilter,
+  SorobanEffectFilter,
+  SubqlBlockHandler,
+  SubqlTransactionHandler,
+  SubqlOperationHandler,
+  SubqlEffectHandler,
 } from '@subql/types-soroban';
 import {plainToClass, Transform, Type} from 'class-transformer';
 import {IsArray, IsEnum, IsInt, IsOptional, IsString, IsObject, ValidateNested} from 'class-validator';
+import {Horizon, ServerApi} from 'stellar-sdk';
 import {SubqlSorobanProcessorOptions} from './types';
 
+export class BlockFilter implements SorobanBlockFilter {
+  @IsOptional()
+  @IsInt()
+  modulo?: number;
+  @IsOptional()
+  @IsString()
+  timestamp?: string;
+}
+
+export class TransactionFilter implements SorobanTransactionFilter {
+  @IsOptional()
+  @IsString()
+  account?: string;
+}
+
+export class OperationFilter implements SorobanOperationFilter {
+  @IsOptional()
+  type: Horizon.OperationResponseType;
+
+  @IsOptional()
+  @IsString()
+  source_account?: string;
+}
+
+export class EffectFilter implements SorobanEffectFilter {
```

Comment:
Structure-wise, contract events and effects are not similar, although both come out of a successful operation. 

the enum EffectType seems to map to numbers instead of the type string. so, we will have to define our own enum.

https://github.com/stellar/js-stellar-sdk/blob/master/src/types/effects.ts